### PR TITLE
Add fuzzing harness

### PR DESCRIPTION
**Pull request by nathaniel-brough**

### CODE REVIEW

**Comment by Cyan4973** [comment]
I'm rather favorable to this effort, given the increased security coverage offered by oss-fuzz,

note though that the current proposal (this PR) fails CI tests.
This should be fixed first.

File: Makefile [hunk 1]
```diff
@@ -315,6 +316,12 @@ test-xxhsum-c: xxhsum
 	echo "00000000  test-expects-file-not-found" | ./xxhsum -c -; test $$? -eq 1
 	@$(RM) .test.*
 
+LIB_FUZZING_ENGINE?="-fsanitize=fuzzer"
+ifeq '' '$(findstring clang++,$(CXX))'
```
Comment (Cyan4973) [inline Makefile]: Is this `ifeq` correct ?
I thought it would require parenthesis

**Comment by nathaniel-brough** [comment]
> I'm rather favorable to this effort, given the increased security coverage offered by oss-fuzz,
> 
> note though that the current proposal (this PR) fails CI tests. This should be fixed first.

Great! Working on getting everything green, it's been a while since I've been working with Makefiles so I'm a little rusty.

File: Makefile (hunk 1 above)
Comment (nathaniel-brough) [reply Makefile]: Your probably right. Just fiddling with this to get it working.

**Review by Cyan4973 — APPROVED** [review]

**Pull request merged**